rasa/nlu/classifiers/diet_classifier.py

```diff
@@ -550,37 +557,67 @@ def _check_input_dimension_consistency(self, model_data: RasaModelData) -> None:
                     "must coincide. Check the output dimensions of previous components."
                 )
 
-    def _extract_labels_precomputed_features(
-        self, label_examples: List[Message], attribute: Text = INTENT
+    def _extract_features_from_all_messages(
+        self,
+        messages: List[Message],
+        attribute: Text = INTENT,
+        type: Optional[Text] = None,
+        featurizers: Optional[List[Text]] = None,
     ) -> Tuple[List[FeatureArray], List[FeatureArray]]:
-        """Collects precomputed encodings."""
-        features = defaultdict(list)
+        """Collects and combines the attribute related features from all messages.
 
-        for e in label_examples:
-            label_features = self._extract_features(e, attribute)
-            for feature_key, feature_value in label_features.items():
-                features[feature_key].append(feature_value)
+        Args:
+            messages: list of messages to extract features from
+            attribute: the only attribute which will be considered
+            type: if not None, the other type (sentence/sequence) will be ignored
+            featurizers: the featurizers to be considered
+        Returns:
+            Sequence level features and sentence level features. Each feature contains
+            FeatureArrays with sparse features first.
+        """
+        if (type is not None) and (type not in [SENTENCE, SEQUENCE]):
+            raise ValueError(f"Unknown type {type}")
```

Comment:
Exception type

Reply:
At this point, I think a ValueError is ok. What exception would you suggest here?